src/kernel-install/kernel-install.in

```diff
@@ -103,25 +103,55 @@ layout=
 initrd_generator=
 
 if [ -r "/etc/kernel/install.conf" ]; then
-    . /etc/kernel/install.conf
+    install_conf="/etc/kernel/install.conf"
 elif [ -r "/usr/lib/kernel/install.conf" ]; then
-    . /usr/lib/kernel/install.conf
+    install_conf="/usr/lib/kernel/install.conf"
+else
+    install_conf=
+fi
+
+if [ -n "$install_conf" ]; then
+    [ "$KERNEL_INSTALL_VERBOSE" -gt 0 ] && echo "Reading $install_conf…"
+    . "$install_conf"
+    # FIXME: This may override configuration in environment variables, e.g. $BOOT_ROOT.
```

Comment:
Wouldn't an (admittedly) ugly solution to this be
```suggestion
    # Sourcing $install_conf may override configuration in environment variables, e.g. $BOOT_ROOT.
    _BOOT_ROOT_FROMENV="$BOOT_ROOT"
    . "$install_conf"
    if [ -n "$_BOOT_ROOT_FROMENV" ] && [ "$BOOT_ROOT" != "$_BOOT_ROOT_FROMENV" ]; then
        BOOT_ROOT="$_BOOT_ROOT_FROMENV"
    fi
```
and so on for every other variable that should be thusly protected?